packages/react-components/react-color-picker-preview/library/docs/Spec.md

```diff
@@ -2,62 +2,458 @@
+| slider | `input` | `input` | Input for slider                                                 |
+| rail   | `div`   | `div`   | It is used to visibly display the min and max selectable values. |
+
+### AlphaSlider
+
+| Property     | Values                   | Default      | Purpose                                |
+| ------------ | ------------------------ | ------------ | -------------------------------------- |
+| customStep   | `number`                 | 3            | Fast step for the slider               |
+| max          | `number`                 | 100          | The max value of the Slider.           |
+| min          | `number`                 | 0            | The min value of the Slider.           |
+| onChange     | `function`               | undefined    | Callback called when color is selected |
+| orient       | `horizontal`, `vertical` | `horizontal` | Orientation of a slider                |
+| overlayColor | `string`                 | undefined    | Overlay color                          |
+| shape        | `square`, `rounded`      | `square`     | Sets shape                             |
+| step         | `number`                 | 1            | Step for the slider                    |
+| value        | `string`                 |              | value of the slider                    |
+
+| Slots  | Values  | Default | Description                                                      |
+| ------ | ------- | ------- | ---------------------------------------------------------------- |
+| root   | `div`   | `div`   | The root of the AlphaSlider element                              |
+| thumb  | `div`   | `div`   | The draggable icon used to select a given value from the Slider. |
+| slider | `input` | `input` | Input for slider                                                 |
+| rail   | `div`   | `div`   | It is used to visibly display the min and max selectable values. |
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### Components
+
+| Component   | Purpose                                      |
+| ----------- | -------------------------------------------- |
+| ColorPicker | Renders ColorPicker                          |
+| ColorArea   | Renders two-dimensional gradient background. |
+| ColorSlider | Renders individual color channel             |
+| AlphaSlider | Renders slider with alpha channel            |
+
+#### ColorPicker component
+
+![ColorPicker Anatomy](./assets/color-picker-anatomy.jpg)
+
+#### DOM
+
+```HTML
+<div role="group" class="fui-ColorPicker" arial-label="ColorPicker">
+  <!-- Content rendered here -->
+</div>
+```
+
+#### ColorArea component
+
+![ColorArea Anatomy](./assets/color-area-anatomy.jpg)
+
+#### DOM
+
+```HTML
+<div role="group" class="fui-ColorArea" aria-label="Saturation and brightness" aria-valuetext="Saturation 54 brightness 35">
```

Comment:
The `aria-valuetext` would need to be on the actual range inputs, since a `role=group` element isn't a form field and can't have a value (and doesn't get focus). I think you actually don't need `aria-valuetext` in this case, since the values are numerical and the `value` should be enough.

For the names, I'd lean towards giving this element (the group one) an `aria-label="gradient area"` or something similar (names are hard 😅), and the two inputs below should have `aria-label="saturation"` and `aria-label="brightness"`